src/components/app/SingleStrategy/GenLender.tsx

```diff
@@ -27,99 +26,106 @@ type GenLenderProps = {
-                    strategy.address,
-                    network
-                );
-                setGenLenderData(loadedGenLenderData);
-                setIsLoading(false);
-            } catch (e: unknown) {
-                console.log('Error:', e);
-                setIsLoading(false);
-                setError(getError(e));
-            }
-        };
-        loadGenLenderData();
-    }, [strategy.address, network]);
+    const { data, loading, error } = useStrategyGenLender(
+        network,
+        strategy.address
+    );
+
+    const getLowestApr = (
+        lendStatuses: LendStatus[],
+        lowestApr: BigNumber,
+        lowestIdx: BigNumber
+    ): BigNumber => {
+        // If there are no assets lent, the lowestApr returned may be
+        // unreasonably high and we should use the actual lender value
+        const lenderLowest = lendStatuses[Number(lowestIdx)][2];
+        if (Number(lowestApr) > Number(lenderLowest)) {
+            return lenderLowest;
+        }
+        return lowestApr;
+    };
+
+    const getLenderName = (
+        lendStatuses: LendStatus[],
+        index: BigNumber
+    ): string => {
+        if (Number(index) >= lendStatuses.length) {
+            return '';
+        }
+        const name = lendStatuses[Number(index)][0];
+        return `${name} (${index.toString()})`;
+    };
 
     const renderData = () => {
-        const lenderStatuses =
-            genLenderData && genLenderData.lendStatuses ? (
-                <>
-                    {genLenderData.lendStatuses.map((value, index) => (
-                        <Box
-                            key={`lender_${value[0]}`}
-                            sx={{
-                                marginBottom:
-                                    index ===
-                                    genLenderData.lendStatuses.length - 1
-                                        ? '0'
-                                        : '20px',
-                            }}
-                        >
-                            <div>
-                                <div>{value[0]}</div>
-                                <EtherScanLink
-                                    address={value[3]}
-                                    network={network}
-                                />
-                            </div>
-                            <div>
-                                {'  '}Deposits:{' '}
-                                {displayAmount(
-                                    value[1].toString(),
-                                    strategy.token.decimals
-                                )}
-                            </div>
-                            <div>
-                                APR: {displayAprAmount(value[2].toString())}
-                            </div>
-                        </Box>
-                    ))}
-                </>
-            ) : (
-                ''
-            );
+        const lenderStatuses = data ? (
```

Comment:
It appears we have a bunch of ternary operators here checking if we have `data`, could we just return `null` if there's no `data` vs having all this logic and assigning empty strings to constants?

Reply:
Yeah good idea. Do you mind if I do this in a follow up? We have this pattern in a bunch of other locations as well, so I'd rather clean it all up at the same time